### PR TITLE
[YUNIKORN-1655] refactor: used yunikorn-scheduler-interface DomainYuniKorn instead of hardcoded string

### PR DESCRIPTION
**Pull request by stu01509**
### What is this PR for?
Used `yunikorn-scheduler-interface` `DomainYuniKorn` instead of hardcoded string.

### What type of PR is it?
* [ ] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [x] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
[YUNIKORN-1655](https://issues.apache.org/jira/browse/YUNIKORN-1655)

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Comment by steinsgateted** [comment]
Hello @stu01509,
Thanks for the PR.
I think we can reuse DomainYuniKorn in scheduler-interface, including separator.
Like:
```
import "github.com/apache/yunikorn-scheduler-interface/lib/go/common"
const DomainYuniKorn = common.DomainYuniKorn
const AnnotationApplicationID = DomainYuniKorn + "app-id"
```
Then I found that not only `common/constants/constants.go` is reused `yunikorn.apache.org/`, like here [queue_quota_mgmt_test.go](https://github.com/apache/yunikorn-k8shim/blob/5fe1b952bd5eb88bf21f49914216b9d49312cec0/test/e2e/queue_quota_mgmt/queue_quota_mgmt_test.go#L42)

**Comment by stu01509** [comment]
Hello @steinsgateted 

Thanks for your review, I updated the commit.

**Comment by steinsgateted** [comment]
Thanks. There are other files that reuse `yunikorn.apache.org/`. Maybe we can change them all.

**Comment by stu01509** [comment]
Hello @steinsgateted 

Thanks for your advice, I replaced the all hardcoded string, please help me to review it.

File: test/e2e/framework/helpers/k8s/pod_annotation.go [hunk 1]
```diff
@@ -19,6 +19,7 @@
 package k8s
 
 import "github.com/apache/yunikorn-k8shim/pkg/appmgmt/interfaces"
+import "github.com/apache/yunikorn-scheduler-interface/lib/go/common"
```
Comment (stu01509) [inline test/e2e/framework/helpers/k8s/pod_annotation.go:22]: Just wanna to confirm with you, Should I replace the `json` content?
```go
type PodAnnotation struct {
	TaskGroupName          string                 `json:"yunikorn.apache.org/task-group-name,omitempty"`
	TaskGroups             []interfaces.TaskGroup `json:"-"`
	SchedulingPolicyParams string                 `json:"yunikorn.apache.org/schedulingPolicyParameters,omitempty"`
	Other                  map[string]string      `json:"-"`
}
```

**Review by steinsgateted — CHANGES_REQUESTED** [review]
@stu01509 Thanks.
In other places maybe we can use `constants.DomainYuniKorn`
There is no need to import interface
Like `pkg/cache/task.go`

**Comment by stu01509** [comment]
Hello @steinsgateted @wilfred-s 

update the commit, but I don't have permission to trigger the CI, could you help me?

**Comment by stu01509** [comment]
Hi @craigcondit, @wilfred-s

Could you help me to trigger the CI?

File: pkg/cache/task.go [hunk 2]
```diff
@@ -435,7 +436,7 @@ func (task *Task) postTaskBound() {
 			if pod.Annotations == nil {
 				pod.Annotations = make(map[string]string)
 			}
-			pod.Annotations["yunikorn.apache.org/scheduled-at"] = strconv.FormatInt(time.Now().UnixNano(), 10)
+			pod.Annotations[siCommon.DomainYuniKorn+"scheduled-at"] = strconv.FormatInt(time.Now().UnixNano(), 10)
```
Comment (steinsgateted) [inline pkg/cache/task.go:439]: Using `constants.DomainYuniKorn` not import SI

File: test/e2e/admission_controller/admission_controller_test.go [hunk 3]
```diff
@@ -34,9 +34,11 @@ import (
 	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/common"
 	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/k8s"
 	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/yunikorn"
+
+	siCommon "github.com/apache/yunikorn-scheduler-interface/lib/go/common"
```
Comment (steinsgateted) [inline test/e2e/admission_controller/admission_controller_test.go:38]: Using `constants.DomainYuniKorn` not import SI

**Review by steinsgateted — CHANGES_REQUESTED** [review]

File: test/e2e/admission_controller/admission_controller_test.go (hunk 3 above)
Comment (stu01509) [reply test/e2e/admission_controller/admission_controller_test.go]: Thank you, update the commit.

**Comment by steinsgateted** [comment]
LGTM

**Review by steinsgateted — APPROVED** [review]

**Review by zhuqi-lucas — APPROVED** [review]
LGTM

**Pull request closed without merging**